Source/CesiumRuntime/Private/CesiumGeoreference.cpp

```diff
@@ -121,28 +113,184 @@ ACesiumGeoreference::GetDefaultGeoreference(const UObject* WorldContextObject) {
   return pGeoreference;
 }
 
-ACesiumGeoreference::ACesiumGeoreference()
-    : AActor(),
-      _ellipsoidRadii{
-          CesiumGeospatial::Ellipsoid::WGS84.getRadii().x,
-          CesiumGeospatial::Ellipsoid::WGS84.getRadii().y,
-          CesiumGeospatial::Ellipsoid::WGS84.getRadii().z},
-      _geoTransforms() {
-  PrimaryActorTick.bCanEverTick = true;
+FVector ACesiumGeoreference::GetOriginLongitudeLatitudeHeight() const {
+  return FVector(OriginLongitude, OriginLatitude, OriginHeight);
+}
 
-  this->Root = CreateDefaultSubobject<USceneComponent>(TEXT("Root"));
-  this->RootComponent = this->Root;
+void ACesiumGeoreference::SetOriginLongitudeLatitudeHeight(
+    const FVector& targetLongitudeLatitudeHeight) {
+  this->_setGeoreferenceOrigin(
+      targetLongitudeLatitudeHeight.X,
+      targetLongitudeLatitudeHeight.Y,
+      targetLongitudeLatitudeHeight.Z);
+}
 
-#if WITH_EDITOR
-  this->SetIsSpatiallyLoaded(false);
-#endif
-  this->SubLevelSwitcher =
-      CreateDefaultSubobject<UCesiumSubLevelSwitcherComponent>(
-          "SubLevelSwitcher");
+FVector ACesiumGeoreference::GetOriginEarthCenteredEarthFixed() const {
+  return UCesiumWgs84Ellipsoid::
+      LongitudeLatitudeHeightToEarthCenteredEarthFixed(
+          this->GetOriginLongitudeLatitudeHeight());
+}
+
+void ACesiumGeoreference::SetOriginEarthCenteredEarthFixed(
+    const FVector& TargetEarthCenteredEarthFixed) {
+  this->SetOriginLongitudeLatitudeHeight(
+      UCesiumWgs84Ellipsoid::EarthCenteredEarthFixedToLongitudeLatitudeHeight(
+          TargetEarthCenteredEarthFixed));
+}
+
+EOriginPlacement ACesiumGeoreference::GetOriginPlacement() const {
+  return this->OriginPlacement;
+}
+
+void ACesiumGeoreference::SetOriginPlacement(EOriginPlacement NewValue) {
+  this->OriginPlacement = NewValue;
+  this->UpdateGeoreference();
+}
+
+double ACesiumGeoreference::GetOriginLatitude() const {
+  return this->OriginLatitude;
+}
+
+void ACesiumGeoreference::SetOriginLatitude(double NewValue) {
+  this->OriginLatitude = NewValue;
+  this->UpdateGeoreference();
+}
+
+double ACesiumGeoreference::GetOriginLongitude() const {
+  return this->OriginLongitude;
+}
+
+void ACesiumGeoreference::SetOriginLongitude(double NewValue) {
+  this->OriginLongitude = NewValue;
+  this->UpdateGeoreference();
+}
+
+double ACesiumGeoreference::GetOriginHeight() const {
+  return this->OriginHeight;
+}
+
+void ACesiumGeoreference::SetOriginHeight(double NewValue) {
+  this->OriginHeight = NewValue;
+  this->UpdateGeoreference();
+}
+
+double ACesiumGeoreference::GetScale() const { return this->Scale; }
+
+void ACesiumGeoreference::SetScale(double NewValue) {
+  if (NewValue < 1e-6) {
+    this->Scale = 1e-6;
```

Comment:
Sadly the Unreal Header Tool isn't smart enough to handle constants (or even #defines) in a UPROPERTY declaration. It reports: `error : Metadata value for 'UIMin' is non-numeric : 'CESIUM_MINIMUM_SCALE'
Even exponential notation doesn't work (1e-6 instead of 0.000001) 😢

But I made this check you've reference in the code use a constant, at least.

Reply:
Thanks